samples/csharp/getting-started/DatabaseIntegration/DatabaseIntegration/Program.cs

```diff
@@ -0,0 +1,102 @@
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms.Categorical;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+
+namespace DatabaseIntegration
+{
+    public class Program
+    {
+        private static string Download(string baseGitPath, string dataFile)
+        {
+            using (WebClient client = new WebClient())
+            {
+                client.DownloadFile(new Uri($"{baseGitPath}"), dataFile);
+            }
+
+            return dataFile;
+        }
+
+        public static string DownloadAdultDataset()
+            => Download("https://raw.githubusercontent.com/dotnet/machinelearning/244a8c2ac832657af282aa312d568211698790aa/test/data/adult.train", "adult.txt");
+
+        public static List<AdultCensus> GetAdultData()
+        {
+            string adultDataset = DownloadAdultDataset();
+            string dataLine;
+            var dataset = new List<AdultCensus>();
+            using (StreamReader reader = new StreamReader(adultDataset))
+            {
+                // read the header
+                dataLine = reader.ReadLine();
+                while ((dataLine = reader.ReadLine()) != null)
+                {
+                    if (string.IsNullOrEmpty(dataLine))
+                        continue;
+
+                    var data = dataLine.Split(',');
+                    dataset.Add(new AdultCensus()
+                    {
+                        Age = int.Parse(data[0]),
+                        Workclass = data[1],
+                        Education = data[3],
+                        //EducationNum = int.Parse(data[4]),
+                        MaritalStatus = data[5],
+                        Occupation = data[6],
+                        Relationship = data[7],
+                        Race = data[8],
+                        Sex = data[9],
+                        CapitalGain = data[10],
+                        CapitalLoss = data[11],
+                        HoursPerWeek = int.Parse(data[12]),
+                        NativeCountry = data[13],
+                        Label = int.Parse(data[14])
+                    });
+                }
+            }
+
+            return dataset;
+        }
+
+        public static void PopulateDatabase()
+        {
+            var adultData = GetAdultData();
+            using (var db = new AdultCensusContext())
+            {
+                foreach(var row in adultData)
+                {
+                    db.AdultCensus.Add(row);
+                }
+
+                var count = db.SaveChanges();
+                Console.WriteLine($"Save results {count}");
+            }
+        }
+
+        public static void Main()
+        {
+            PopulateDatabase();
+            using(var db = new AdultCensusContext())
+            {
+                var mlContext =  new MLContext();
+                var dataView = mlContext.Data.ReadFromEnumerable(db.AdultCensus);
```

Comment:
use db.AdultCensus.AsNoTracking() here to get a stream of objects. Otherwise the DbContext will unnecessarily start tracking every object returned by the query, and all the data will need to fit in memory at the same time. 

I get that this is a bit artificial in this case because we hold all the data in a `List<T>`, but I think it is still important to recommend [no tracking queries](https://docs.microsoft.com/en-us/ef/core/querying/tracking) in case customers want to read larger datasets using EF Core.

Reply:
I wonder if you could do something a bit more interesting here, e.g. add a .Where() operator that allows selecting only some subset of the data from the command line. Or random sorting on the database (assuming random sorting is useful in this case).

Reply:
This makes me wonder if there would be value in having a method in the example that encapsulates how to get an IDataView from and EF Core query, e.g. pushing the AsNoTracking() and the optional ability to do some randomization there.

Reply:
Also left feedback in another comment about NoTracking.

---
In reply to: [270256412](https://github.com/dotnet/machinelearning-samples/pull/295#discussion_r270256412) [](ancestors = 270256412)

Reply:
So what I have now is that we query all the data from the table and use that for training. Then query again only selecting the top 2000 rows to use for prediction. Does this seem reasonable? Or are you wanting something more precise in that we always select one portion of the data for training and another portion for testing?

---
In reply to: [270259895](https://github.com/dotnet/machinelearning-samples/pull/295#discussion_r270259895) [](ancestors = 270259895)